src/IssueViz.Security.UnitTests/Taint/TaintIssueToIssueVisualizationConverterTests.cs

```diff
@@ -78,16 +90,12 @@ public void Convert_FromSonarQubeIssue_IssueVizConverterCalledWithCorrectParamet
             false);
 
         var expectedConvertedIssueViz = CreateIssueViz();
-        var issueVizConverter = Substitute.For<IAnalysisIssueVisualizationConverter>();
         issueVizConverter.Convert(Arg.Any<IAnalysisIssueBase>(), null)
             .Returns(expectedConvertedIssueViz);
 
-        var testSubject = CreateTestSubject(issueVizConverter);
         var result = testSubject.Convert(taintDto, "C:\\root");
 
         result.Should().BeSameAs(expectedConvertedIssueViz);
-
-        expectedConvertedIssueViz.Received().IsSuppressed = true;
         issueVizConverter.Received().Convert(
             Arg.Is((TaintIssue taintIssue) =>
                 taintIssue.IssueKey == "serverkey" &&
```

Comment:
Please extract the checks for PrimaryLocation and flow locations in two helper methods. This monstrous test is difficult to follow, check and maintain. 

Example: 
```
 private static bool HasExpectedLocation(IAnalysisIssueLocation location, TaintLocationDto dto) =>
     location.Message == dto.message &&
     location.FilePath == dto.filePath &&
     location.TextRange.LineHash == dto.textRange.hash &&
     location.TextRange.StartLine == dto.textRange.startLine &&
     location.TextRange.EndLine == dto.textRange.endLine &&
     location.TextRange.StartLineOffset == dto.textRange.startLineOffset &&
     location.TextRange.EndLineOffset == dto.textRange.endLineOffset;
```

Reply:
This lambda is an Expression and it doesn't work with method calls. The argument matcher fails.